### PR TITLE
[proxima-direct-io-kafka] #203 fix IllegalStateException in offset seek

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptor.java [hunk 1]
```diff
@@ -449,11 +449,11 @@ private void seekConsumerToBeginning(ConsumerId consumerId, Collection<TopicPart
     }
 
     private <K, V> ConsumerRecords<K, V> pollConsumer(
-        ConsumerGroup group,
-        long period,
-        ConsumerId consumerId,
-        ElementSerializer<K, V> serializer,
-        @Nullable ConsumerRebalanceListener listener)
+        final ConsumerGroup group,
```
Comment (dmvk) [inline direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptor.java:452]: why this change?

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java [hunk 2]
```diff
@@ -392,14 +393,17 @@ private void submitConsumerWithObserver(
                 log.debug(
                     "End offsets of current assignment {}: {}", kafka.assignment(), endOffsets);
               }
-
-              listener.onPartitionsAssigned(kafka.assignment());
             } while (poll.isEmpty()
+                && numPolls++ < 2
```
Comment (dmvk) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:397]: can you please increment the counter inside the cycle, so the construct is more human readable?

Comment (je-ik) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java]: Hm, why is this not readable? I think this is pretty common pattern. I can do that, of course, but I don't think that will result in better readability.

File: direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptor.java (hunk 1 above)
Comment (je-ik) [reply direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptor.java:452]: You mean the `final`? The `consumerId` is used for synchronization inside the method, so it should be final, and when doing that I made all parameters final for consistency.

**Comment by dmvk** [comment]
can you please add a failing test case for the issue?

**Comment by je-ik** [comment]
Adding a test is not very meaningful. Generally, assignment is not well defined until rebalance listerner onPartitionsAssigned is called. That is why I moved the logic there and the positive tests, confirming it works should suffice.
